src/Common.php

```diff
@@ -1,143 +1,127 @@
 <?php
+
+declare(strict_types=1);
+
 namespace Keboola\OAuthV2Api;
 
+use Closure;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Exception\RequestException;
 use GuzzleHttp\HandlerStack;
 use GuzzleHttp\Middleware;
-use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Request;
+use JsonException;
+use Keboola\OAuthV2Api\Exception\ClientException;
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
+use Throwable;
 
-/**
- *
- */
 class Common
 {
-    /**
-     * @var string
-     */
-    protected $apiUrl = 'https://syrup.keboola.com/oauth-v2/';
+    protected Client $client;
+    private const DEFAULT_BACKOFF_RETRIES = 10;
+    private const DEFAULT_HEADERS = ['Accept' => 'application/json'];
 
     /**
-     * @var ClientWrapper
+     * @param array{url: string, handler?: HandlerStack, backoffMaxTries?: int} $config
+     * @throws ClientException
      */
-    protected $client;
-
-    /**
-     * @var int
-     */
-    protected static $backOffMaxRetries = 10;
-
-    /**
-     * @var array
-     */
-    protected $defaultHeaders = [
-        'Accept' => 'application/json'
-    ];
-
-    /**
-     * @var bool
-     */
-    protected $returnArrays = false;
```

Comment:
jinak se to nepoužívalo, takže jsem to zjednodušil, že to vždy vrací asociativní pole